### PR TITLE
WIP: demo of using rollup code splitting prototype rule

### PR DESCRIPTION
**Pull request by gregmagolan**

### CODE REVIEW

**Comment by kylecordes** [comment]
If I understand right from the bazel rule in here, it is hardcoded to know about the entry point for the lazy loaded foo module.  I recently also used rollup to create a working set of lazy bundles, with that same manual gathering of the entry points. I'm really looking forward to seeing something to gather them automatically - from some chat discussion it sounds like the "official" way to do this could take a few months to come through.

**Comment by gregmagolan** [comment]
@kylecordes Yes. This is an early prototype for this and the entry points are hard-coded in the rule and in index.html.

**Comment by alexeagle** [comment]
Igor and I have been working here:
https://github.com/angular/angular/pull/21431

in an earlier snapshot, there was an `ng_entry_point` rule that you could put in another package - but it only provided string data so it was simpler to just list them in the bundle.

Longer-term we should get them from a static analysis output from individual libraries.

File: src/index.html [hunk 1]
```diff
@@ -17,17 +18,32 @@
...
     <script src="/zone.min.js"></script>
 
-    <!--
-      For development mode, this script will be bundled on-the-fly by
-      the devserver, based on the latest JS files built by Bazel.
-      For production, you should be able to use the same index.html file, but
-      this JS resource would be served from a static file.
+    <script src="/system.js"></script>
+
+    <script>
```
Comment (alexeagle) [inline src/index.html]: looks like this script could easily be another output of the rollup_bundle rule, placed in the runfiles.transitive_files so it's available to be loaded by anything with a transitive data[] dep

**Comment by alexeagle** [comment]
Seems like some files are not git add'ed? Where's the change in `rollup_bundle.bzl`?

**Comment by kylecordes** [comment]
@gregmagolan @alexeagle Thanks - I will keep an eye on that PR and any successors, and try to provide a bit of testing-with-code-out-here-in-the-wild where feasible.

**Comment by gregmagolan** [comment]
> https://github.com/bazelbuild/rules_nodejs/pull/104/files

@alexeagle You mean the one in here? https://github.com/bazelbuild/rules_nodejs/pull/104/files :)

**Comment by alexeagle** [comment]
Closing this PR since we don't plan to merge right now, the primary copy of this code is in Greg's fork

**Pull request closed without merging**